src/libosd/include/osd/cl_cdm.h

```diff
@@ -0,0 +1,129 @@
+/* Copyright 2018 The Open SoC Debug Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef OSD_CL_CDM_H
+#define OSD_CL_CDM_H
+
+#include <osd/module.h>
+#include <osd/osd.h>
+#include <osd/packet.h>
+#include <osd/hostmod.h>
+
+#include <stdlib.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @defgroup libosd-cl_cdm CDM client
+ * @ingroup libosd
+ *
+ * @{
+ */
+
+/**
+ * Information about a Core Debug Module
+ */
+struct osd_cdm_desc {
+    unsigned int di_addr; //!< DI address of CDM module 
+    uint16_t core_ctrl; //!< Control signal for CPU core
+    uint16_t core_reg_upper; //!< Most significant bits of the SPR address 
+    uint16_t core_data_width; //!< CPU data width in bits
+};
+
+/**
+ * Read data from a special-purpose register of the CPU Core attached to a 
+ * Core Debug Module (CDM)
```

Comment:
The first (and only first) line of the docblock is the summary, then follows a newline and then the long description.